From 2b1df4bc7dd7a3d622e77d846868faa22caca824 Mon Sep 17 00:00:00 2001
From: Magne Sjaastad <magne.sjaastad@ceetronsolutions.com>
Date: Mon, 1 Apr 2019 14:26:15 +0200
Subject: [PATCH] #4268 libecl : Check for nullpointer on Windows

---
 ThirdParty/Ert/lib/util/util.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ThirdParty/Ert/lib/util/util.c b/ThirdParty/Ert/lib/util/util.c
index 53a0c8584..6243a411f 100644
--- a/ThirdParty/Ert/lib/util/util.c
+++ b/ThirdParty/Ert/lib/util/util.c
@@ -2153,6 +2153,11 @@ int util_fmove( FILE * stream , long offset , long shift) {
 #ifdef HAVE_WINDOWS__ACCESS
 
 bool util_access(const char * entry, int mode) {
+  if (!entry)
+  {
+    return false;
+  }
+
   return (_access(entry, mode) == 0);
 }
 
-- 
2.21.0.windows.1

